### PR TITLE
added release, restructure code pretty heavily

### PR DESCRIPTION
**Pull request by jmccarthy14**
@iproctor @jwolski @ryansokol @whoevergetsachancetolook

Take a look.  I need to keep working on tests but release functionality should be added in as well as some bugs i saw while in the code.  README  update incoming also.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,60 +1,29 @@
...
 
-/**
- * Checks each VNode to see if the current node owns it, and if it does it
- * prompts recovery of each key.  For example, it finds that it owns VNode B,
- * recovers 14 keys that the old owner of VNode B was working on, and prompts
- * the client via callback to recover each of those keys, leaving that to the
- * individual client's business logic.
- * @param {function} done The callback when all keys have been loaded.
- */
-Sevnup.prototype.loadAllKeys = function loadAllKeys(done) {
```
Comment (iproctor) [inline index.js]: you're still calling loadAllKeys in line 50

Comment (iproctor) [inline index.js]: u need some tests for this stuff

File: lib/vnode-store.js [hunk 2]
```diff
@@ -84,6 +93,77 @@ VNodeStore.prototype.removeKeyFromVNode = function removeKeyFromVNode(vnodeName,
...
+        }
+    ],
+    function (err, results) {
+        done(err, results);
+    });
+};
+
+VNodeStore.prototype._releaseVNodes = function releaseVNodes(vnodes, done) {
+    var self = this;
+    self._handleVNodes(vnodes, self.releaseVNodeKeys, done);
```
Comment (iproctor) [inline lib/vnode-store.js]: self.releaseVNodeKeys.bind(this)

File: lib/vnode-store.js [hunk 3]
```diff
@@ -84,6 +93,77 @@ VNodeStore.prototype.removeKeyFromVNode = function removeKeyFromVNode(vnodeName,
...
+};
+
+VNodeStore.prototype._recoverVNodes = function recoverVNodes(vnodes, done) {
+    var self = this;
+    self._handleVNodes(vnodes, self.loadVNodeKeys, done);
```
Comment (iproctor) [inline lib/vnode-store.js]: bind(this)

File: lib/vnode-store.js [hunk 4]
```diff
@@ -84,6 +93,77 @@ VNodeStore.prototype.removeKeyFromVNode = function removeKeyFromVNode(vnodeName,
...
+ * @param {array} vnodes The vnodes to handle.
+ * @param {function} handler The handler that takes a vnodeName and callback as
+ *     parameters.
+ * @param {function} done The callback to call when all vnodes are handled.
+ */
+VNodeStore.prototype._handleVNodes = function handleVNodes(vnodes, handler, done) {
+    async.eachLimit(
+        vnodes,
+        MAX_PARALLEL_TASKS,
+        function (vnode, eachDone) {
```
Comment (iproctor) [inline lib/vnode-store.js]: can just pass handler here

Comment (jmccarthy14) [inline lib/vnode-store.js]: haha chru

**Pull request merged**